lib/file_download.rb

```diff
@@ -0,0 +1,36 @@
+require "net/http"
+
+# Used to download a remote file. Handles redirects.
+# Returns the file as a `Tempfile`
+class FileDownload
+  attr_reader :url
+
+  def initialize(url)
+    @url = url
+  end
+
+  def fetch
+    download_file(url)
+  end
+
+  private
+
+  def download_file(url)
+    response = Net::HTTP.get_response(URI(url))
+
+    case response
+      when Net::HTTPSuccess
```

Comment:
At a risk of bikeshedding over style - is it OK if we unindent the `when`s here? Everywhere else we've used `case` statements, the `when`s have been indented at the same level as the `case`, and it just (IMO) makes the statement a bit more readable.

Reply:
👍🏽